testsuite/ext-tools/hub-examples/install-pkg-all-systems.py

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/python3
+import xmlrpc.client
+import argparse
+from datetime import datetime, timezone
+
+# Parsing arguments
+parser = argparse.ArgumentParser()
+parser.add_argument('hub_fqdn', type=str, help='The Hub FQDN')
+parser.add_argument('user', type=str, help='The username for the Hub')
+parser.add_argument('password', type=str, help='The password for the Hub')
+parser.add_argument('packageId', type=int, help='Package ID')
+args = parser.parse_args()
+
+api = f"http://{args.hub_fqdn}:2830/hub/rpc/api"
+client = xmlrpc.client.ServerProxy(api)
+
+print("Login with autoconnect mode...")
+loginResponse = client.hub.loginWithAutoconnectMode(args.user, args.password)
+hubSessionKey = loginResponse["SessionKey"]
+print(f"hubSessionKey: {hubSessionKey}\n")
+
+print("Get the server IDs...")
+serverIds = client.hub.listServerIds(hubSessionKey)
+print(f"serverIds: {serverIds}\n")
+
+print("Multicast call to get the list of systems per server IDs...")
+systemsPerServer = client.multicast.system.list_systems(hubSessionKey, serverIds)
+successfulResponses = systemsPerServer["Successful"]
+failedResponses = systemsPerServer["Failed"]
```

Comment:
I don't see the reason why we need to have 4 files with this fragment repeated. I would extend `parser` to have this `packageId` argument optional, and add an argument of what kind of authentication it would be, something like that:

```
parser.add_argument('packageId', type=int, nargs='?', default=None, help='Package ID (optional)')
parser.add_argument('login_type', type=str, choices=['autoconnect', 'credentials', 'relay'], help='The type of login to use')
```

Reply:
Package ID is not optional, if you don't provide one... you can't achieve the goal of the script (see filename)

Reply:
As those are examples and not a tool, I prefer to have separated examples, to make them more clear.